src/index.js

```diff
@@ -639,6 +639,19 @@ const members = {
       while (parsed.hasChildNodes()) {
         const firstChild = parsed.firstChild;
 
+        // when we are doing this: root.innerHTML = "<slot><div></div></slot>";
+        // slot.__childNodes is out of sync with slot.childNodes.
+        // to fix it we have to manually remove and insert them
+        if (canPatchNativeAccessors && getNodeType(firstChild) === 'slot' && (firstChild.__childNodes.length !== firstChild.childNodes.length)) {
+          while (firstChild.hasChildNodes()) {
+            firstChild.removeChild(firstChild.firstChild);
+          }
+
+          for (let i = 0; i < firstChild.__childNodes.length; i++) {
+            firstChild.appendChild(firstChild.__childNodes[i]);
+          }
```

Comment:
It might be good to move this into a function with a descriptive name, like `synchroniseChildNodes` or something like that. That way it's a bit clearer what the point of the code block is – I'd probably still leave the comment though, as it's a bit strange that we need to do it.